### PR TITLE
(MODULES-5222) - managevim option added to manifests. 

### PR DESCRIPTION
**Pull request by david22swan**
When set to false .vim file is not created. Defaults to true.

e.g.

```
accounts::user { 'hunner':
            groups               => ['root'],
            password             => 'hi',
            shell                => '/bin/true',
            home                 => '/test/hunner',
            home_mode            => '0700',
            managevim            => false,
            bashrc_content       => file('accounts/shell/bashrc'),
            bash_profile_content => file('accounts/shell/bash_profile'),
            sshkeys              => [
              'ssh-rsa #{test_key} vagrant',
              'command="/bin/echo Hello",from="myhost.example.com,192.168.1.1" ssh-rsa #{test_key} vagrant2'
            ],
          }
```

### CODE REVIEW

File: manifests/home_dir.pp [hunk 1]
```diff
@@ -40,6 +43,7 @@
 define accounts::home_dir(
   String $user,
   String $group,
+  Boolean $managevim                     = undef,
```
Comment (pmcmaw) [inline manifests/home_dir.pp]: Need to be set or be Optional.

Comment (HelenCampbell) [inline manifests/home_dir.pp]: Needs to be Optional if it can be undef.

File: manifests/user.pp [hunk 2]
```diff
@@ -221,6 +225,7 @@
       groups         => $groups,
       membership     => $membership,
       managehome     => $managehome,
+      # managevim      => $managevim,
```
Comment (pmcmaw) [inline manifests/user.pp]: shouldn't be commented out.

File: manifests/user.pp [hunk 3]
```diff
@@ -237,6 +242,7 @@
...
```
Comment (pmcmaw) [inline manifests/user.pp]: shouldn't be commented out.

**Pull request merged**